framework/src/utils/FillBetweenPointVectorsTools.C

```diff
@@ -558,8 +563,12 @@ isBoundaryOpenSingleSegment(ReplicatedMesh & mesh,
 }
 
 bool
-isExternalBoundary(ReplicatedMesh & mesh, const boundary_id_type bid)
+isExternalBoundary(MeshBase & mesh, const boundary_id_type bid)
 {
+  // This has no communication and expects elem_ptr to find any
+  // element, so it only works on serialized meshes
+  MeshSerializer serial(mesh);
```

Comment:
is this a no-op on an already serialized mesh?

Reply:
Yup.

The one catch w.r.t. performance is that, if you're calling a bunch of functions like this in a row, you want the mesh to be already serialized, not rely on the serializer, because otherwise the serializer will happily serialize and then unserialize and serialize and unserialize as many times as you ask it to.

In the long run we might want some MOOSE interface for users to specify "don't bother distributing anything while we do all this stitching and such" vs "okay, we're about to extrude 100x more elements and then refine to get 8^n more, *definitely* make sure we're distributed first".  Maybe one MeshGenerator to turn off distribution and another to force it, or maybe an option in a base class parameter?